src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -2213,6 +2237,131 @@ METHOD(kernel_ipsec_t, destroy, void,
 	free(this);
 }
 
+struct tunnel_counter_key {
+	uint32_t terminator_id;
+	char* tenant_id;
+	char* account_id;
+	char* vrf_name;
+	uint64_t count; /*Stores the current tunnel count value.
+					* Compare/hash function don't consider it is not part of
+					* the identification of the tunnel*/
+};
+
+static bool tunnel_counter_key_compare(const void *key, const void *other_key) {
+    const struct tunnel_counter_key *tunnel_1 = (const struct tunnel_counter_key *)key;
+    const struct tunnel_counter_key *tunnel_2 = (const struct tunnel_counter_key *)other_key;
+
+	return tunnel_1->terminator_id == tunnel_2->terminator_id &&
+			streq(tunnel_1->tenant_id, tunnel_2->tenant_id) && 
+			streq(tunnel_1->account_id, tunnel_2->account_id) && 
+			streq(tunnel_1->vrf_name, tunnel_2->vrf_name);
+}
+
+static u_int tunnel_counter_key_hash(const void *key) {
+	const struct tunnel_counter_key *tunnel_key = (const struct tunnel_counter_key *)key;
+	return chunk_hash_inc(chunk_from_thing(tunnel_key->terminator_id),
+						  chunk_hash_inc(chunk_from_str(tunnel_key->tenant_id),
+						  chunk_hash_inc(chunk_from_str(tunnel_key->account_id),
+						  chunk_hash(chunk_from_str(tunnel_key->vrf_name)))));
+}
+
+static struct tunnel_counter_key *create_new_tunnel_counter_key(uint32_t terminator_id,
+																const char* tenant_id,
+																const char* account_id,
+																const char* vrf_name){
+
+	struct tunnel_counter_key *key = (struct tunnel_counter_key *)malloc(sizeof(struct tunnel_counter_key));
+
+	key->terminator_id = terminator_id;
+	key->tenant_id = strdup(tenant_id);
+	key->account_id = strdup(account_id);
+	key->vrf_name = strdup(vrf_name);
+	key->count = 1;
```

Comment:
For such initializations use the `INIT()` macro

Reply:
Now that this function is so thin and only calls a single function macro, better to just add the call to `INIT()` directly in `write_terminator_tunnels_oper()` and remove this function